test/envtest/konnect_entities_cacertificate_test.go

```diff
@@ -0,0 +1,154 @@
+z/csoAAnfMBY401j2NNUgPpwx5sTQdCz5aFDmanol5152M8CAwEAAaNTMFEwHQYD
+VR0OBBYEFK2qd3oRF37acVvgfDeLakx66ioTMB8GA1UdIwQYMBaAFK2qd3oRF37a
+cVvgfDeLakx66ioTMA8GA1UdEwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggEB
+AAuul+rAztaueTpPIM63nrS4bSZsIatCgAQ5Pihm0+rZ+13BJk4K2GxkS+T0qkB5
+34+F3eVhUB4cC+kVkWZrlEzD9BsJwWjnoJK+848znTg+ufTeaOQWslYNqFKjmy2k
+K6NE7E6r+JLdNvafJzeDybSTXI1tCzDRWUdj5m+bgruX07B13KIJKrAweCTD1927
+WvvfJYxsg8P7dYD9DPlcuOm22ggAaPPu4P/MsnApiq3kJEI/nSGSsboKyjBO2hcz
+VF1CYr6Epfyw/47kwuJLCVHjlTgT4haOChW1S8rZILCLXfb8ukM/g3XVYIeEwzsr
+KU74cm8lTFCdxlcXePbMdHc=
+-----END CERTIFICATE-----
+`
+
+func TestKongCACertificate(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := Context(t, context.Background())
+	defer cancel()
+	cfg, ns := Setup(t, ctx, scheme.Get())
+
+	t.Log("Setting up the manager with reconcilers")
+	mgr, logs := NewManager(t, ctx, cfg, scheme.Get())
+	factory := ops.NewMockSDKFactory(t)
+	sdk := factory.SDK
+	reconcilers := []Reconciler{
+		konnect.NewKonnectEntityReconciler(factory, false, mgr.GetClient(),
+			konnect.WithKonnectEntitySyncPeriod[configurationv1alpha1.KongCACertificate](konnectInfiniteSyncTime),
+		),
+	}
+	StartReconcilers(ctx, t, mgr, logs, reconcilers...)
+
+	t.Log("Setting up clients")
+	cl, err := client.NewWithWatch(mgr.GetConfig(), client.Options{
+		Scheme: scheme.Get(),
+	})
+	require.NoError(t, err)
+	clientNamespaced := client.NewNamespacedClient(mgr.GetClient(), ns.Name)
+
+	t.Log("Creating KonnectAPIAuthConfiguration and KonnectGatewayControlPlane")
+	apiAuth := deployKonnectAPIAuthConfigurationWithProgrammed(t, ctx, clientNamespaced)
+	cp := deployKonnectGatewayControlPlaneWithID(t, ctx, clientNamespaced, apiAuth)
+
+	t.Log("Setting up SDK expectations on KongCACertificate creation")
+	sdk.CACertificatesSDK.EXPECT().CreateCaCertificate(mock.Anything, cp.GetKonnectStatus().GetKonnectID(),
+		mock.MatchedBy(func(input sdkkonnectcomp.CACertificateInput) bool {
+			return input.Cert != nil &&
+				*input.Cert == dummyValidCACertPEM
+		}),
+	).Return(&sdkkonnectops.CreateCaCertificateResponse{
+		CACertificate: &sdkkonnectcomp.CACertificate{
+			ID: lo.ToPtr("12345"),
+		},
+	}, nil)
+
+	t.Log("Setting up a watch for KongCACertificate events")
+	w := setupWatch[configurationv1alpha1.KongCACertificateList](t, ctx, cl, client.InNamespace(ns.Name))
+
+	t.Log("Creating KongCACertificate")
+	createdCert := &configurationv1alpha1.KongCACertificate{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "cert-",
+			Namespace:    ns.Name,
+		},
+		Spec: configurationv1alpha1.KongCACertificateSpec{
+			ControlPlaneRef: &configurationv1alpha1.ControlPlaneRef{
+				Type: configurationv1alpha1.ControlPlaneRefKonnectNamespacedRef,
+				KonnectNamespacedRef: &configurationv1alpha1.KonnectNamespacedRef{
+					Name: cp.GetName(),
+				},
+			},
+			KongCACertificateAPISpec: configurationv1alpha1.KongCACertificateAPISpec{
+				Cert: dummyValidCACertPEM,
+				Tags: []string{"tag1", "tag2"},
+			},
+		},
+	}
```

Comment:
Maybe we create a deploy func for this in `test/envtest/deploy_resources.go`?

Reply:
243ba44